linera-chain/src/data_types.rs

```diff
@@ -801,55 +794,70 @@ impl HashedCertificateValue {
     }
 }
 
-// TODO(#1987): Consider refactoring BlockProposal to make this unnecessary.
-#[derive(Debug, Serialize, Deserialize)]
-pub struct ProposalPayload<'a> {
-    pub content: Cow<'a, BlockAndRound>,
-    pub outcome: Option<Cow<'a, BlockExecutionOutcome>>,
+/// The data a block proposer signs: the round, the block, and, if it is a retry from an earlier
+/// consensus round, the oracle records.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ProposalContent {
+    pub block: Block,
+    pub round: Round,
+    pub oracle_records: Option<Vec<OracleRecord>>,
```

Comment:
Could we document what `None` vs `Some<vec![]>` mean here/how the consumer is expected to interpret them?

Reply:
The comment gives a hint but perhaps the field could be named `forced_oracle_records` to make it clearer.

Reply:
Or `recorded_oracle_responses`?

As part of (or as a follow-up to) https://github.com/linera-io/linera-protocol/pull/2060 (once I've fixed the issue with the tests) I was meaning to replace `OracleRecord` with `Vec<OracleResponse>` anyway. If that's okay I'd add the documentation when I do that, since it's more related to that than to this PR?

Reply:
One PR at a time. Here, we expect `something_oracle_records`.

Can you move the documentation of the fields that you already wrote (`the round, the block, and, if it is a retry from an earlier consensus round, the oracle records`) to where it belongs?